### PR TITLE
Mutatuion testing

### PR DESCRIPTION
**Pull request by S1x6**
Убил все предложенные мутации кроме последних 2х, для предпоследней есть тест, но он не работает, последний вообще не писал
+ пару рандомных

### CODE REVIEW

File: src/test/kotlin/qbit/StorageTest.kt [hunk 1]
```diff
@@ -39,4 +43,43 @@ class StorageTest {
...
+
+        val root = Files.createTempDirectory("qbit").toFile()
+        val storage = FileSystemStorage(root)
+
+        val conn = qbit(storage)
+        conn.persist(_id)
+
+        val e = Entity(_id eq "1")
+        val trx = conn.trx()
+        trx.persist(e)
```
Comment (d-r-q) [inline src/test/kotlin/qbit/StorageTest.kt]: @S1x6 Is it essential for mutant elimination to persist via explicit transaction, but not implicit transaction within conn.persist?

File: src/test/kotlin/qbit/model/EntityTest.kt [hunk 2]
```diff
@@ -82,6 +82,14 @@ class EntityTest {
         assertEquals("s2", e[ref][s])
     }
 
+    @Test
+    fun testEmptyRefList() {
+        val ref = RefListAttr(root["ref"], false)
+        val e = AttachedEntity(EID(0), mapOf(ref to ArrayList<EID>()), emptyDb, false)
```
Comment (d-r-q) [inline src/test/kotlin/qbit/model/EntityTest.kt]: Such behavior (empty reflist) is impossible with current storage format
It's better to add check, that list is not empty in AttachedEntity.tryGet and expect AssertionError in this test

File: src/test/kotlin/qbit/StorageTest.kt [hunk 3]
```diff
@@ -39,4 +43,43 @@ class StorageTest {
...
+        trx.commit()
+
+        val loaded = storage.load(Namespace("refs")["head"])
+        val hash = conn.db.hash.bytes
+        Assert.assertArrayEquals(loaded, hash)
+    }
+
+    @Ignore
```
Comment (d-r-q) [inline src/test/kotlin/qbit/StorageTest.kt]: Move this test into separate branch, so other changes may be merged

File: build.gradle [hunk 4]
```diff
@@ -32,6 +33,14 @@ compileTestKotlin {
 }
 
 apply plugin: 'maven'
+apply plugin: 'info.solidsoft.pitest'
+
+pitest {
+    targetClasses = ['qbit.*']  //by default "${project.group}.*"
+    threads = 4
+    outputFormats = ['XML', 'HTML']
```
Comment (d-r-q) [inline build.gradle]: Remove XML format, if it will not break anything

File: src/test/kotlin/qbit/StorageTest.kt (hunk 1 above)
Comment (S1x6) [reply src/test/kotlin/qbit/StorageTest.kt]: Changed to implicit transaction

File: src/test/kotlin/qbit/model/EntityTest.kt (hunk 2 above)
Comment (S1x6) [reply src/test/kotlin/qbit/model/EntityTest.kt]: Temporarily removed this test

File: src/test/kotlin/qbit/StorageTest.kt (hunk 3 above)
Comment (S1x6) [reply src/test/kotlin/qbit/StorageTest.kt]: Moved

File: build.gradle (hunk 4 above)
Comment (S1x6) [reply build.gradle]: Done

**Pull request merged**